File: python_ruby/oop_variable/3.py
class Calc(object):
    def __init__(self,v1,v2):
        if isinstance(v1, int):
            self.v1 = v1
        if isinstance(v2, int):
            self.v2 = v2
    def add(self):
        return self.v1 + self.v2

    def subtract(self):
        return self.v1 - self.v2

    def setV1(self, v1):
        if isinstance(v1, int):
            self.v1 = v
    def getV1(self):
        return self.v1

c = Calc(10,10)
print(c.add())
print(c.subtract())

c.setV1('one')
c.v2 = 30
print(c.add())
print(c.subtract())

File: python_ruby/oop_variable/3.rb
class Calc
  attr_reader :v1,:v2
  attr_writer :v1
  def initialize(v1,v2)
    @v1 = v1
    @v2 = v2
  end
  def add()
    return @v1+ @v2
  end

  def subtract()
    return @v1 - @v2
  end
  def setV1(v)
    if v.is_a?(Integer)
      @v1 = v
    end
   end
  def getV1()
    return @v1
  end
end

c = Calc.new(10,10)
p c.add()
p c.subtract()
c.setV1('one')
c.v1 = 20
p c.add()
p c.getV1()
